solr/core/src/java/org/apache/solr/handler/UpdateRequestHandler.java

```diff
@@ -123,6 +128,31 @@ public void init(NamedList<?> args) {
     loaders = Collections.unmodifiableMap(createDefaultLoaders(args));
   }
 
+  @Override
+  public void initializeMetrics(SolrMetricsContext parentContext, String scope) {
+    super.initializeMetrics(parentContext, scope);
+    metricsLocal =
+        new HandlerMetrics(
+            solrMetricsContext, getCategory().toString(), scope + LOCAL_HANDLER_SUFFIX);
+  }
+
+  @Override
+  public HandlerMetrics getMetricsForThisRequest(SolrQueryRequest req) {
+    // using exactly the same logic in 8.x to preserve the behavior
+    boolean isDistrib =
+        req.getParams()
+            .getBool(
+                CommonParams.DISTRIB,
+                req.getCore() != null
```

Comment:
do we need zk check here? as zk will be always there.

Reply:
Should be rather safe this is copied from 8.x RequestHandlerBase#handleRequest :) https://github.com/fullstorydev/lucene-solr/blob/release/8.8/solr/core/src/java/org/apache/solr/handler/RequestHandlerBase.java#L209

Reply:
do you think this check is helping here? if `distrib` prop is not there, and request is on right node, then it will tell distributed request?

Reply:
I tried to dig in the actual code logic yesterday and it's quite complicated. 😅 

To be honest I don't think this metrics works as expected in 8.x. `distrib` does not seem to be used by the update flow at all (it might use it to make some GET requests, but it does not affect how update is handled). From reference check, it only seems to apply for `SearchHandler`/`RealTimeGetFlow` flow.

Also by debugging the update flow, on a high level it goes:
1. Update requests first reaches UpdateRequestHandler, which streams the input docs via a processor chain
2. Towards the end of the chain. It's the `DistributedZkUpdateProcessor`, which has a setupRequest method: https://github.com/fullstorydev/lucene-solr/blob/release/8.8/solr/core/src/java/org/apache/solr/update/processor/DistributedZkUpdateProcessor.java#L596 . This is the method that figure out whether the current node can handle this request or not, and set a non null `nodes` if re-direct is needed
3. It usually stops here for us, because the core is the correct one (we map the core before sending updates). However, if it should redirect to another node (ie nodes is not empty), it will re-distribute the request to the correct node at https://github.com/fullstorydev/lucene-solr/blob/release/8.8/solr/core/src/java/org/apache/solr/update/processor/DistributedZkUpdateProcessor.java#L275 . Take note that it does NOT set `distrib` here, it sets some other internal distrib flags but it's not `distrib`

Therefore in 8.x, the "local" update metric has the highest hit, as for normal "correct" updates, we add `distrib=false`, though such distrib flag was probably not really used for the actual update logic, except the "local" metrics itself. Now "distributed" update metric almost worked "accidentally" to catch those "wrong" updates, because the first update has `distrib=false`, but if `DistributedZkUpdateProcessor` determines it should re-distribute it, it will distribute the request without `distrib`, and that triggered requests is now counted as "distributed" as it does not define `distrib=false` 😁 

So my conclusion after all the investigation - for 9.x, lets just keep the same metrics behavior as 8.x for now, it is very tricky to get it right (that's why they didn't do it at first place in 9.x I believe)